test/unit/package_manager_base_test.py

```diff
@@ -14,6 +14,9 @@ def setup(self):
         repository.root_dir = 'root-dir'
         self.manager = PackageManagerBase(repository)
 
+    def teardown(self):
+        del self.manager
```

Comment:
Yep agree that `del` should be avoided if possible. I will dig if I can manage a better test for that, in fact I expected that GC would do that for me in the test, maybe I can rearrange the scope.